core/trino-main/src/test/java/io/trino/sql/planner/TestLogicalPlanner.java

```diff
@@ -1332,6 +1332,32 @@ public void testRemovesTrivialFilters()
                         values("nationkey", "name", "regionkey", "comment")));
     }
 
+    @Test
+    public void testRemoveTrivialFiltersWithPartitioningHandle()
```

Comment:
The PartitioningHandle [issue](https://github.com/trinodb/trino/issues/21506) has been fixed by https://github.com/trinodb/trino/pull/27191.
We still want to add this optimization so that empty union branches are pruned. Please rename the test to `testRemoveEmptyUnionBranch` or similar.